prov/rxm/src/rxm_conn.c

```diff
@@ -1462,15 +1462,9 @@ static int rxm_conn_atomic_progress_eq_cq(struct rxm_ep *rxm_ep,
 	memset(entry, 0, RXM_MSG_EQ_ENTRY_SZ);
 
 	while(1) {
-		/* Comply with restricted threading levels that MSG provider
-		 * may have been configured with */
-		if (rxm_ep->msg_info->domain_attr->threading != FI_THREAD_SAFE)
-			rxm_ep->cmap->acquire(&rxm_ep->cmap->lock);
-
+		ofi_ep_lock_acquire(&rxm_ep->util_ep);
 		again = fi_trywait(rxm_fabric->msg_fabric, fids, 2);
-
-		if (rxm_ep->msg_info->domain_attr->threading != FI_THREAD_SAFE)
-			rxm_ep->cmap->release(&rxm_ep->cmap->lock);
+		ofi_ep_lock_release(&rxm_ep->util_ep);
```

Comment:
This is changing to the ep lock, so the commit message is confusing.  A prior commit already killed cmap lock, didn't it?

Reply:
Ah, Github rearranged the commit order based on commit date instead of the real order. It always does this. :/

I can modify the dates if it makes it easier to review. Here is the correct order:

```
67225c8 prov/rxm: add an env variable that defines the default data progress mode
2999cc4 prov/rxm: fix missing error handling for FI_GETWAIT
6c07329 prov/verbs: serialize rdma_cm_get_event and rdma_destroy_ep
d887bd5 prov/rxm: add a unspec CM thread notification event
644254a prov/util: define new ep_list_lock for accessing av->ep_list
4d9130d Revert "prov/rxm: fix serialization for atomics auto-progress"
e9e15ea prov/rxm: replace all cmap->locks with ep->lock and add any missing serialization
3218321 prov/rxm: reject connection request in shutdown state
eb3cd74 prov/rxm: improve logging + debug asserts
```

Reply:
I'm going to see if I can file a bug against github for this.  It makes it difficult to review PRs by looking at individual commits.

Reply:
github knows this.  They just don't care.  There's a work-around for this, but it's a pain.  The problem shows up when doing git rebase.

Reply:
What's the workaround? Even if we can detect that the chronological and topological order don't match, we can manually correct it.

The issue is a minor annoyance but it can catch people off-guard.

Reply:
I consider it more than a minor annoyance.  :)  See this link for a work-around.  It's a manual thing that everyone would need to do when running git rebase and updating a PR.

https://stackoverflow.com/questions/44704519/how-to-fix-commit-order-in-github-pull-requests-broken-by-git-rebase